docs/src/references/c8y-configuration-management.md

```diff
@@ -0,0 +1,162 @@
+(which respective SmartRest2 codes are `526` and `524`).
+
+```shell
+$ ls -l /etc/tedge/operations/c8y/c8y_UploadConfigFile
+-rw-r--r-- 1 tedge tedge 95 Mar 22 14:24 /etc/tedge/operations/c8y/c8y_UploadConfigFile
+
+$ cat /etc/tedge/operations/c8y/c8y_UploadConfigFile
+[exec]
+  topic = "c8y/s/ds"
+  on_message = "526"
+  command = "/usr/bin/c8y_configuration_plugin"
+  
+$ ls -l /etc/tedge/operations/c8y/c8y_DownloadConfigFile
+-rw-r--r-- 1 tedge tedge 97 Mar 22 14:24 /etc/tedge/operations/c8y/c8y_DownloadConfigFile
+
+$ cat /etc/tedge/operations/c8y/c8y_DownloadConfigFile
+[exec]
+  topic = "c8y/s/ds"
+  on_message = "524"
+  command = "/usr/bin/c8y_configuration_plugin"
+```
+
+## Configuration
+
+The `c8y_configuration_plugin` configuration is stored by default under `/etc/tedge/c8y/c8y_configuration_plugin.toml`
+
+This [TOML](https://toml.io/en/) file defines the list of files to be managed from the cloud tenant.
+
+```shell
+$ cat /etc/tedge/c8y/c8y_configuration_plugin.toml
+files = [
+    '/etc/tedge/tedge.toml',
+    '/etc/tedge/mosquitto-conf/c8y-bridge.conf',
+    '/etc/tedge/mosquitto-conf/tedge-mosquitto.conf',
+    '/etc/mosquitto/mosquitto.conf'
+  ]
+```
+
+Note that:
+* The file `/etc/tedge/c8y/c8y_configuration_plugin.toml` itself doesn't need to be listed.
+  This is implied, so the list can *always* been configured from the cloud.
+* If the file `/etc/tedge/c8y/c8y_configuration_plugin.toml`
+  is not found, empty, ill-formed or not-readable
+  then only `c8y_configuration_plugin.toml` is managed from the cloud.
+* If the file `/etc/tedge/c8y/c8y_configuration_plugin.toml` is ill-formed
+  or cannot be read then an error is logged, but the operation proceed
+  as if the file were empty.
+  
+The behavior of the `c8y_configuration_plugin` is also controlled
+by the configuration of thin-edge:
+
+* `tedge config get mqtt.bind_address`: the address of the local MQTT bus.
+* `tedge config get mqtt.port`: the TCP port of the local MQTT bus.
+* `tedge config get tmp.path`: the directory where the files are updated
+  before being copied atomically to their targets.
+* `tedge config get log.path`: the directory used for logging.
+
+## Usage
+
+```shell
+$ c8y_configuration_plugin
+Thin-edge device configuration management for Cumulocity
+
+USAGE:
+    c8y_configuration_plugin [OPTIONS] [SMARTREST_OPERATION]
+
+OPTIONS:
+        --config-file <CONFIG_FILE>    [default: /etc/tedge/c8y/c8y_configuration_plugin.toml]
+    -h, --help                         Print help information
+    -V, --version                      Print version information
+
+SMARTREST_OPERATION:
+    The SmartRest2 message received on `c8y/s/ds` and triggering either a download or an upload.
+    This operation is then processed in its enterity by the `c8y_configuration_plugin`,
+    notifying the Cumulocity tenant of its progress (messages `501`, `502` and `503`).
+    
+    If no SmartRest2 operation is provided, `c8y_configuration_plugin`
```

Comment:
To avoid the plugin doing MQTT stuff, why not have an `--init` option which will make `c8y_configuration_plugin` print all the supported files on to the console? And let the mapper send the `119` MQTT message with the supported files read from the console.